Jvollmer init

jrvollmer opened the pull request.

jrvollmer committed:
TODO Add CMakeLists

TODO README:
- Add notes about dependencies (e.g. mbed cli 2 and arm gcc - probably
  just point to mbed docs on setting up mbed cli 2)
- Possibly expand on the initial README in terms of purpose

jrvollmer committed:
TODO Add workflow to compile and flash project

TODO
Add note on README about BTF integration

- Uses `git reflog` to determine if any *.lib files have changed between
  workflow runs. If not, mbed-os/ is not modified between workflow runs.
- Uses `mbed-tools detect` to get information about the device connected
  to the pi. The target specified in the setup configuration file is
  used to verify that the correct target is connected.
- Runs the BTF within a fresh Python virtual environment.

jrvollmer committed:
TODO Add setup and tests templates

TODO Add test/ directtory and instructions for adding a test file,
including a test in testRunner.cpp, and addin test info/paths to the
configuration files

jrvollmer committed:
TODO Generate definitions from configuration files

TODO
- Add a section to the README about the yaml parsing package used by the
  Python script

jrvollmer committed:
TODO Add Makefile

TODO
Add run instructions to README
- Include compile and flash targets

jrvollmer commented:
Note for when this is actually ready for review: Set this up with Mbed Studio to see if the `.gitignore` contents get reordered. I have a stash entry that reorders `.gitignore`

jrvollmer commented:
TODOs:
- [x] Integrate with the [BTF](https://github.com/badgerloop-software/BTF)
- [ ] README updates (see commit messages)
- [x] Drop the most recent commit ("TODO Add tentative test/setup information")
- [x] Set this up with Mbed Studio to see if the `.gitignore` contents get reordered. I have a stash entry that reorders `.gitignore`

Wigu701 commented on mbed-os.lib:
```diff
@@ -0,0 +1 @@
+https://github.com/ARMmbed/mbed-os/#17dc3dc2e6e2817a8bd3df62f38583319f0e4fed
```
Consider changing this to a fixed version (Firmware running 6.17.0 for 2023-2024) to prevent major revision update debacles if possible here. If not, consider building it into the library update workflow

jrvollmer replied:
Isn't that a fixed version? From my understanding, it points to a specific commit hash. I can look into which version it is though and call it out.

jrvollmer replied:
Okay, so I checked the mbed-os repo, and the commit hash in mbed-os.lib points to the [6.17.0 release](https://github.com/ARMmbed/mbed-os/releases/tag/mbed-os-6.17.0). This should be fine

Wigu701 commented on parseConfigs.py:
```diff
@@ -0,0 +1,68 @@
+				keep = False
+		elif line == '# ---- END MACROS ----\n':
+			cmake.append(line)
+			keep = True
```
May be a cleaner solution to read in the entire file, then use regex to substitute the macro block.

jrvollmer replied:
I've switched the whole process to generating a header file from scratch with the definitions. It should be a much cleaner solution

Wigu701 commented on setup.yml:
```diff
@@ -0,0 +1,32 @@
+#     <driver>:
+#       <attribute>: <value>
+#
+# NOTE: If <value> is an array, format it as a Python list (i.e. [...])
```
Instructions and templates should be documented in a README somewhere, either here if for this workflow or in BTF if relating to mock side.

jrvollmer replied:
Moved to README

Wigu701 commented on tests.yml:
```diff
@@ -0,0 +1,17 @@
+# --- Test Options ---
+# mock: Raspberry Pi emulates the device through software
+# hardware: Hardware setup connected to the UUT. The Raspberry Pi does not mock the device
+# disabled: Test will not be run
```
Same comment as with setup.yml

Instructions and templates should be documented in a README somewhere, either here if for this workflow or in BTF if relating to mock side.

jrvollmer replied:
Moved to README

jrvollmer committed:
TODO Starter make instructions README.md

jrvollmer committed:
TODO test execution starter

TODO
- Split into serialWrapper (first) and test execution (second) commits
- Make Ethan a co-author of the seriaLWrapper commit:
https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors

jrvollmer committed:
TODO Squash

TODO This has major changes to serialWrapper

jrvollmer committed:
TODO modify test execution and results

jrvollmer committed:
TODO Fixup serial test
